File: server/controllers/account_controller.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from typing import List
from database import get_db
from services.account_service import AccountService
from schemas.account import AccountResponse, AccountSummary

router = APIRouter(prefix="/api", tags=["accounts"])


@router.get("/user/{user_id}/accounts", response_model=List[AccountResponse])
def get_accounts(user_id: str, db: Session = Depends(get_db)):
    """Get all accounts for a user."""
    service = AccountService(db)
    accounts = service.get_accounts_by_user_id(user_id)
    return accounts


@router.get("/user/{user_id}/accounts/summary", response_model=AccountSummary)
def get_account_summary(user_id: str, db: Session = Depends(get_db)):
    """Get account summary with totals."""
    service = AccountService(db)
    summary = service.get_account_summary(user_id)
    return summary


@router.get("/user/{user_id}/accounts/{account_type}", response_model=AccountResponse)
def get_account_by_type(user_id: str, account_type: str, db: Session = Depends(get_db)):
    """Get account by type."""
    service = AccountService(db)
    account = service.get_account_by_type(user_id, account_type)
    return account


@router.get("/account/{account_id}", response_model=AccountResponse)
def get_account(account_id: int, db: Session = Depends(get_db)):
    """Get account details."""
    service = AccountService(db)
    account = service.get_account(account_id)
    return account


@router.get("/account/{account_id}/balance")
def get_balance(account_id: int, db: Session = Depends(get_db)):
    """Get account balance only."""
    service = AccountService(db)
    balance = service.get_balance(account_id)
    return {"balance": balance}

File: server/controllers/accountController.js
const AccountService = require('../services/accountService');

const AccountController = {
  // GET /api/user/:userId/accounts
  getAccounts(req, res) {
    try {
      const accounts = AccountService.getAccountsByUserId(req.params.userId);
      res.json(accounts);
    } catch (error) {
      if (error.message.includes('not found')) {
        res.status(404).json({ error: error.message });
      } else {
        res.status(500).json({ error: error.message });
      }
    }
  },

  // GET /api/user/:userId/accounts/summary
  getAccountSummary(req, res) {
    try {
      const summary = AccountService.getAccountSummary(req.params.userId);
      res.json(summary);
    } catch (error) {
      if (error.message.includes('not found')) {
        res.status(404).json({ error: error.message });
      } else {
        res.status(500).json({ error: error.message });
      }
    }
  },

  // GET /api/account/:id
  getAccount(req, res) {
    try {
      const account = AccountService.getAccount(parseInt(req.params.id));
      res.json(account);
    } catch (error) {
      if (error.message.includes('not found')) {
        res.status(404).json({ error: error.message });
      } else {
        res.status(500).json({ error: error.message });
      }
    }
  },

  // GET /api/user/:userId/accounts/:type
  getAccountByType(req, res) {
    try {
      const account = AccountService.getAccountByType(req.params.userId, req.params.type);
      res.json(account);
    } catch (error) {
      if (error.message.includes('not found')) {
        res.status(404).json({ error: error.message });
      } else {
        res.status(500).json({ error: error.message });
      }
    }
  },

  // GET /api/account/:id/balance
  getBalance(req, res) {
    try {
      const balance = AccountService.getBalance(parseInt(req.params.id));
      res.json({ balance });
    } catch (error) {
      if (error.message.includes('not found')) {
        res.status(404).json({ error: error.message });
      } else {
        res.status(500).json({ error: error.message });
      }
    }
  }
};

module.exports = AccountController;
